xeofs/preprocessing/sanitizer.py

```
@@ -85,17 +86,30 @@ def transform(self, X: DataArray) -> DataArray:
         # Check if input has the correct coordinates
         self._check_input_coords(X)
 
-        # Store sample coordinates for inverse transform
-        self.sample_coords_transform = X.coords[self.sample_name]
-        # Remove NaN entries; only consider full-dimensional NaNs
-        # We already know valid features from the fitted dataset
-        X = X.isel({self.feature_name: self.is_valid_feature})
-        # However, we need to recheck for valid samples, as the new dataset may
-        # have different samples
-        is_valid_sample = ~X.isnull().all(self.feature_name).compute()
-        X = X.isel({self.sample_name: is_valid_sample})
-        # Store valid sample locations for inverse transform
-        self.is_valid_sample_transform = is_valid_sample
+        X_valid_features = self._get_valid_features(X)
+
+        # Remove full-dimensional NaN entries
+        X = X.dropna(dim=self.sample_name, how="all")
+        X = X.dropna(dim=self.feature_name, how="all")
+
+        # For new data only, validate that NaN features match the original
+        if self.has_run:
+            if not X_valid_features.equals(self.is_valid_feature):
```

Couldn't decide whether this check should also go under the `if not dask` conditional. Any thoughts?

---

In its current form, it would force a computation even with Dask, right? In that case, I'd vote to apply it only to non-Dask data. We should just make it clear in the documentation (preferably [here](https://xeofs.readthedocs.io/en/latest/overview_2_features.html#handling-missing-values)) that transform automatically checks for `NaNs`, but only for non-Dask arrays.

---

Correct, but not on the first `fit_transform()` call, only on a second run of `transform()`. Let me implement a test to see how this actually plays out if we pass incorrect features and let this error out later. As long as that is somewhat intelligible what went wrong (or we can make it so by catching and expanding on the right error) I think your suggestion is good.